rubix-presto/src/main/java/com/qubole/rubix/presto/PrestoClusterManager.java

```diff
@@ -175,18 +165,17 @@ public void initialize(Configuration conf)
               // keep only the healthy nodes
               allNodes.removeAll(failedNodes);
 
-              Set<String> hosts = new HashSet<String>();
+              List<ClusterNode> hosts = new ArrayList<>();
 
               for (Stats node : allNodes) {
-                hosts.add(node.getUri().getHost());
+                hosts.add(new ClusterNode(node.getUri().getHost(), NodeState.ACTIVE));
               }
               if (hosts.isEmpty()) {
                 // case of master only cluster
-                hosts.add(InetAddress.getLocalHost().getHostAddress());
+                hosts.add(new ClusterNode(InetAddress.getLocalHost().getHostAddress(), NodeState.ACTIVE));
               }
-              List<String> hostList = Lists.newArrayList(hosts.toArray(new String[0]));
-              Collections.sort(hostList);
-              return hostList;
+
+              return hosts;
```

Comment:
We are not sorting the list anymore and neither do we do it when when use it in bookkeeper. Wouldn't this cause inconsistency because node indexes are not guaranteed to be same for the same membership of cluster.